scripts/lightweight_curator.py

```diff
@@ -98,33 +99,40 @@ def get_actionable_indices(es, max_allowed_size, index_name_prefixes_list):
     """
     Returns a list of actionable indices based on percentage size and age.
     """
-    class data_structure:
+    class indices_struct:
       def __init__(self, name, size, creation_date):
         self.name = name
         self.size = size
         self.creation_date = creation_date
 
-    # Prepare dictionary of indices with their size and creation_date values.
-    size_counter = 0
-    data_array = []
-
+    """
+    Appends all indice into the list with their name, size and creation_date values.
+    """
+    indices = []
     for index_name_prefixes in index_name_prefixes_list:
         for name in es.indices.get_alias(index=index_name_prefixes + "*").keys():
             size = int(get_first_item(es.indices.stats(index=name)['indices'][name]['total']['store']))
             creation_date = int(es.indices.get(index=name)[name]['settings']['index']['creation_date'])
-            data_array.append(data_structure(name, size, creation_date))
+            indices.append(indices_struct(name, size, creation_date))
 
-    # Output are one or more indices which are above the 80% threshold and are supposed to be deleted.
+    """
+    Returns list of indices which are above threshold limit.
+    """
     indices_to_delete = []
-    for data_object in sorted(data_array, key=lambda x: x.creation_date, reverse=True):
-        if size_counter < max_allowed_size and data_object.size + size_counter < max_allowed_size:
-            log(log_info, "Do not add into actionable list: '{indice}', summed disk usage is {usage} B and disk limit is {limit} B".format(
-                indice=data_object.name, usage=size_counter, limit=int(max_allowed_size)))
-            size_counter += data_object.size
+    indices_size_counter = 0
+    for index in sorted(indices, key=lambda x: x.creation_date, reverse=True):
+        expected_size = index.size + indices_size_counter
+        def index_smaller_then_max_allowed_size(indices_size_counter, expected_size, max_allowed_size):
+            if indices_size_counter < max_allowed_size and expected_size < max_allowed_size:
+                return True
+        if index_smaller_then_max_allowed_size(indices_size_counter, expected_size, max_allowed_size):
+            log(log_info, "Do not add into actionable list: '{index}', summed disk usage is {usage} B and disk limit is {limit} B".format(
```

Comment:
Might I suggest using f strings if you are using an updated python version

https://www.geeksforgeeks.org/formatted-string-literals-f-strings-python/

Reply:
Done.